### PR TITLE
[BE] Excluded the specified endpoint from mw

### PR DESCRIPTION
**Pull request by YoginiTayade**
# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **New Features**
  - Implemented new middleware to enhance request processing for academic years and programs.
  - Introduced additional route configurations for improved navigation and access control.

- **Enhancements**
  - Improved assignment logic for user identification in service methods.
  - Enhanced user access validation methods for programs and academic years.

- **Refactor**
  - Replaced `CohortMiddleware` with `AcademicYearIdMiddleware` and `ProgramMiddleware` across multiple modules.
  - Updated middleware applications to streamline request handling and access control.

- **Bug Fixes**
  - Corrected user identification assignment to ensure proper user context in requests.

- **Documentation**
  - No visible changes to end-users.

- **Style**
  - No visible changes to end-users.

- **Tests**
  - No visible changes to end-users.

- **Chores**
  - No visible changes to end-users.

- **Revert**
  - No visible changes to end-users.

### CODE REVIEW

File: src/src/user/user.service.ts [hunk 1]
```diff
@@ -27,12 +27,13 @@ export class UserService {
...
 		userId: string,
 		body: any,
-		tableName: String,
 		req: any,
+		tableName: String,
 	) {
 		try {
-			const program_id = req.mw_program_id;
 			const academic_year_id = req.mw_academic_year_id;
+			console.log(academic_year_id);
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:35]: remove

File: src/src/user/user.service.ts [hunk 2]
```diff
@@ -71,13 +72,14 @@ export class UserService {
...
 			let validation_query = `query MyQuery {
-				users(where: {group_users: {user_id: {_eq:${userId}}, member_type: {_eq: "owner"}, status: {_eq: "active"}}, program_faciltators: {academic_year_id: {_eq:${academic_year_id}}, program_id: {_eq:${program_id}}}}) {
+				users(where: {group_users: {user_id: {_eq:${userId}}, member_type: {_eq: "owner"}, status: {_eq: "active"}}, program_faciltators: {academic_year_id: {_eq:${academic_year_id}}}}) {
 				  id
 				}
 			  }
 
 			  `;
-
+			//console.log(validation_query);
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:81]: remove

File: src/src/user/user.service.ts [hunk 3]
```diff
@@ -106,6 +108,8 @@ export class UserService {
 					id: userId,
 				},
 			};
+			console.log(data.query);
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:111]: remove

File: src/src/facilitator/facilitator.service.ts [hunk 4]
```diff
@@ -1738,18 +1741,14 @@ export class FacilitatorService {
 		  aadhar_token
 		  address
 		  aadhar_verified
-		  block_id
```
Comment (manojLondhe) [inline src/src/facilitator/facilitator.service.ts:1741]: are these fields removed on purpose from response?

Comment (YoginiTayade) [reply src/src/facilitator/facilitator.service.ts:1741]: yes. they were unnecessary leading to slow speed of api

File: src/src/user/user.service.ts (hunk 2 above)
Comment (YoginiTayade) [reply src/src/user/user.service.ts]: resolved

File: src/src/user/user.service.ts (hunk 3 above)
Comment (YoginiTayade) [reply src/src/user/user.service.ts]: resolved

**Pull request merged**